lib/game.rb

```diff
@@ -0,0 +1,72 @@
+require_relative "../config/environment.rb"
+
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+  ]
+
+  def current_player
+    board.cells.select{|cell| !cell.strip.empty?}.count.even? ? player_1 : player_2
+  end
+
+  def winner
+    winning_combo = WIN_COMBINATIONS.detect do |combo|
```

Comment:
This method is one that should take advantage of the `won?` method to see if there is a winning combo. If so, it would return the token at either of the 3 positions in the winning combo returned from `won?`

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.